src/Sentry.Unity.Editor.iOS/NativeMain.cs

```diff
@@ -0,0 +1,80 @@
+using System.IO;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal interface INativeMain
+    {
+        public void AddSentry(string pathToMain);
+    }
+
+    internal class NativeMain : INativeMain
+    {
+        private const string Include = @"#include <Sentry/Sentry.h>
+#include ""SentryOptions.m""
+";
+        private const string Init = @"
+        [SentrySDK startWithOptions:getSentryOptions()];
+";
+
+        public void AddSentry(string pathToMain)
+        {
+            if (!DoesMainExist(pathToMain))
+            {
+                return;
+            }
+
+            var main = File.ReadAllText(pathToMain);
+            if (ContainsSentry(main))
+            {
+                return;
+            }
+
+            var sentryMain = AddSentryToMain(main);
+            if (sentryMain is null)
+            {
+                return;
+            }
+
+            File.WriteAllText(pathToMain, sentryMain);
+        }
+
+        internal bool DoesMainExist(string pathToMain)
+        {
+            if (!File.Exists(pathToMain))
+            {
+                Debug.LogWarning($"Could not find '{pathToMain}'.");
+                return false;
+            }
+
+            return true;
+        }
+
+        internal bool ContainsSentry(string main)
+        {
+            if (main.Contains(Include))
+            {
+                Debug.Log("Sentry already added to 'main.mm'.");
```

Comment:
I wonder if we should pass `SentryOptions` in the constructor to this class (or just `IDiagnosticLogger`) via eh constructor so we can use our logger (and its configuration) instead of the Unity logger here?